### PR TITLE
Added static site for the public-apis

### PR DESCRIPTION
**Pull request by pawelborkar**
With reference to #1612 this PR will add a static site for the public-apis with url public-apis.dev
As we all know we have to use GitHub Pages while deploying the site to the mentioned domain after this PR got merged.
But while doing so there's one problem we are unable to enforce the Https in for our domain as some the APIs are not having https in their domain which means their lacking the SSL certification which makes our site to fall in the category of [mixed content](https://docs.github.com/en/pages/getting-started-with-github-pages/securing-your-github-pages-site-with-https).

Here are the screenshots of the website built by using [MKDocs](https://mkdocs.org)

Public-Apis_homepage

Public-Apis_search

Public-Apis_contributepage

Public-Apis_License

### CODE REVIEW

**Comment by zaherg** [comment]
@PawelBorkar  we can test it and if the mixed content caused any issue I can use cloudflare (instead of google) as the DNS management and enforce https using them.

**Comment by zaherg** [comment]
I have setup the required DNS modifications so once you enable/merged this PR the domain should work with no time.

File: mkdocs.yml
Comment (sent) [inline mkdocs.yml:2]:
```suggestion
site_url: http://public-apis.github.io/
```

**Review by sent — CHANGES_REQUESTED** [review]
Don't use a user-provided domain names, there is no guarantee that they just could disappear anytime.

Github.io is the way to go.

**Review by matheusfelipeog — COMMENTED** [review]
Good job @PawelBorkar 😄

But with this file structure we will have to keep the `README.md` file and `docs/index.html` synchronized manually with each new api added. This is repeated for the other files too: `contributions.md`, `license`

I think it is better to use just 1 file to list the APIs and be used as a basis for the site. This is good for us who maintain the project and for the employees. So we avoid duplicate work.

The MKDocs does not allow using the root of the directory. It needs a child directory to structure its files, for example, a `docs` directory. Sorry for indicating MKDocs for this, I didn't know. 😢 

I looked for alternatives to solve this problem and found [Docsify](https://docsify.js.org/) ([Docsify on Github](https://github.com/docsifyjs/docsify))

The [free-for-dev](https://github.com/ripienaar/free-for-dev) project uses it, you can see how it looks here: [https://free-for.dev](https://free-for.dev/#/)

And everything seems to work well.

I did some tests to see it working:

image
image

I just had to make a correction to the `README.md` file to work.

What do you think?

If you like it and allow it, I can do that job.

And again, sorry.

**Comment by marekdano** [comment]
@PawelBorkar Thank you very much for working on putting the project on the website version. I appreciate it!!!

> But with this file structure we will have to keep the README.md file and docs/index.html synchronized manually with each new api added. This is repeated for the other files too: contributions.md

@matheusfelipeog I agree that it'll be duplicate work. But this can be done by having a script that will update the `docs` folder every time before deploying the site.

TBH I'm happy with both options, using MKDocs (doing all manual updates by a script) or Docsify. We should decide what'll involve less maintenance to update the doc.

**Review by marekdano — COMMENTED** [review]
@PawelBorkar Thank you very much for working on putting the project on the website version. I appreciate it!!!

But with this file structure we will have to keep the README.md file and docs/index.html synchronized manually with each new api added. This is repeated for the other files too: contributions.md

@matheusfelipeog I agree that it'll be duplicate work. But this can be done by having a script that will update the docs folder every time before deploying the site.

TBH I'm happy with both options, using MKDocs (doing all manual updates by a script) or Docsify. We should decide what'll involve less maintenance to update the doc.

**Comment by PawelBorkar** [comment]
Thanks for the feedback @matheusfelipeog and for pointing out the limitations of MKDocs and for finding the alternative to it. 
When I was testing the site I do get to know that we have to update the docs/index.md as well whenever we update the README but at that time I thought that we'll write a script that will automate this as @marekdano has [suggested ](https://github.com/public-apis/public-apis/pull/1633#pullrequestreview-638207050) in this way we don't have to bother about the docs/index.md

> The free-for-dev project uses it, you can see how it looks here: https://free-for.dev
> And everything seems to work well

I did check out their website and found some limitations.

- In order to render the README we have to remove all the badges from the README otherwise they too get rendered.
- The search functionality makes it difficult for the reader to read as it gets out of focus from the reader's eyes.


As you can see here

Screenshot (57)

As Compared to 

search

And above all, we need to make the site easy to navigate for the new users/contributors which will increase their user experience and pool in more contributors which can be achieved here as it gives us a navigation bar with all the necessary functionality and search. It's easy to see the contributing guide and license in here as compared to what we are getting with Docsify.

Navigate

We just need to add a script will which pull the changes made in README & Contributing.md and copy it to the docs/index.md and contributing.md.

Feel free to give your suggestions.

Thanks You All :-)

**Review by Tiaogaviao — APPROVED** [review]
Ganhar dinheiro baixando o Kwai!https://s.kwai.app/s/oenFdtwS

**Comment by Tiaogaviao** [comment]
Ganhar dinheiro baixando o Kwai!https://s.kwai.app/s/oenFdtwS

**Comment by pawelborkar** [comment]
Closing this as there are many unofficial and better alternative available now.

**Pull request closed without merging**
